src/Compilers/Core/AnalyzerDriver/DiagnosticDescriptorExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class DiagnosticDescriptorExtensions
+    {
+        public static void SetOnLocalizableStringException(this DiagnosticDescriptor descriptor, DiagnosticAnalyzer owner, Action<Exception, DiagnosticAnalyzer, Diagnostic> onLocalizableStringException)
```

Comment:
it could be nice if we can do what you did for localizable string (exception safe localizable string) to diagnostic description as well.

var exceptionSafeDiagnosticDescriptor = userDiagnosticDescriptor.MakeExceptionSafe(onException);